projects/buttons_and_adc/src/main.c

```diff
@@ -0,0 +1,61 @@
+// Created by Sudhish M on 2020-10-04.
+// buttons_and_adc
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static GpioAddress adc_pin_address{ .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress button_address{ .port = GPIO_PORT_B, .pin = 2 };
```

Comment:
This is C, not C++, so the uniform initialization syntax won't work here. You'll need a `=` between the variable name and the initializer, like `s_adc_pin_address = { ... }`. Also, we prefix static variables with `s_` to make it clear that they're static.